kotlin-grpc/src/test/kotlin/com/linecorp/armeria/server/grpc/CoroutineServerInterceptorTest.kt

```diff
@@ -0,0 +1,100 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.grpc.com.linecorp.armeria.server.grpc
+
+import com.linecorp.armeria.common.RequestContext
+import com.linecorp.armeria.common.grpc.GrpcStatusFunction
+import com.linecorp.armeria.internal.testing.AnticipatedException
+import com.linecorp.armeria.server.ServerBuilder
+import com.linecorp.armeria.server.ServiceRequestContext
+import com.linecorp.armeria.server.auth.Authorizer
+import com.linecorp.armeria.server.grpc.CoroutineServerInterceptor
+import com.linecorp.armeria.server.grpc.GrpcService
+import com.linecorp.armeria.testing.junit5.server.ServerExtension
+import io.grpc.Metadata
+import io.grpc.ServerCall
+import io.grpc.ServerCallHandler
+import io.grpc.Status
+import kotlinx.coroutines.future.await
+import org.junit.jupiter.api.extension.RegisterExtension
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+
+internal class CoroutineServerInterceptorTest {
+    private class AuthInterceptor : CoroutineServerInterceptor {
+        private val authorizer = Authorizer { ctx: ServiceRequestContext, metadata: Metadata ->
+            val future = CompletableFuture<Boolean>()
+            ctx.eventLoop().schedule({
+                if (ctx.request().headers().contains("Authorization", "Bearer token-1234")) {
+                    future.complete(true)
+                } else {
+                    future.complete(false)
+                }
+            }, 100, TimeUnit.MILLISECONDS)
+            return@Authorizer future
+        }
+
+        override suspend fun <ReqT, RespT> suspendedInterceptCall(
+            call: ServerCall<ReqT, RespT>,
+            headers: Metadata,
+            next: ServerCallHandler<ReqT, RespT>
+        ): ServerCall.Listener<ReqT> {
+            val result = authorizer.authorize(ServiceRequestContext.current(), headers).await()
+            if (result) {
+                return next.startCall(call, headers)
+            } else {
+                throw AnticipatedException("Invalid access")
+            }
+        }
+    }
+
+    companion object {
+        @RegisterExtension
+        val server: ServerExtension = object : ServerExtension() {
+            override fun configure(sb: ServerBuilder) {
+                val statusFunction = GrpcStatusFunction { ctx: RequestContext?, throwable: Throwable, metadata: Metadata? ->
+                    if (throwable is AnticipatedException && "Invalid access" == throwable.message) {
```

Comment:
We can now move the constant to the RHS given Kotlin's `==` operator doesn't throw an NPE even if LHS is `null`.
```suggestion
                    if (throwable is AnticipatedException && throwable.message == "Invalid access") {
```

Reply:
In [7f563df](https://github.com/line/armeria/pull/4724/commits/7f563df07affb7bb528eb06ac41da58ede7c28ea), I flipped constant's position 🙂